src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        for (const data in extraData) {
+          stateCopy[data] = extraData[data];
```

Comment:
avoid nested loops. Instead of for () you can use `Object.assign`

Reply:
this is true, you can just use `Object.assign`